app/docs/0.10.x/loadbalancing.md

```diff
@@ -182,6 +184,116 @@ as a single target, with the specified weight. Upon every proxied request
 to this target it will query the nameserver again.
 
 
-[Back to TOC](#table-of-contents)
+[Back to Table of Contents](#table-of-contents)
+
+### **Blue-green deployments**
+
+Using the ring-balancer a blue-green deployment can be easily orchestrated for 
+an api. Switching target infrastructure only requires a `PATCH` request on an
+api, to change the `upstream` name. 
+
+Set up the "Blue" environment, running version 1 of the address service;
+
+```bash
+# create an upstream
+$ curl -X POST http://kong:8001/upstreams \
+    --data "name=address.v1.service"
+
+# add two targets to the upstream
+$ curl -X POST http://kong:8001/upstreams/address.v1.service/targets \
+    --data "target=192.168.34.15:80"
+    --data "weight=100"
+$ curl -X POST http://kong:8001/upstreams/address.v1.service/targets \
+    --data "target=192.168.34.16:80"
+    --data "weight=50"
+
+# create an api targetting the Blue upstream
+$ curl -X POST http://kong:8001/apis/ \
+    --data "name=address-service" \
+    --data "hosts=address.mydomain.com" \
+    --data "upstream_url=http://address.v1.service/address"
+```
+
+Requests with host header set to `address.mydomain.com` will now be proxied
+by Kong to the two defined targets; 2/3 of the requests will go to
+`http://192.168.34.15:80/address` (`weight=100`), and 1/3 will go to
+`http://192.168.34.16:80/address` (`weight=50`).
+
+Before deploying version 2 of the address service set up the "Green"
+environment.
+
+```bash
+# create a new Green upstream for address service v2
+$ curl -X POST http://kong:8001/upstreams \
+    --data "name=address.v2.service"
+
+# add targets to the upstream
+$ curl -X POST http://kong:8001/upstreams/address.v2.service/targets \
+    --data "target=192.168.34.17:80"
+    --data "weight=100"
+$ curl -X POST http://kong:8001/upstreams/address.v2.service/targets \
+    --data "target=192.168.34.18:80"
+    --data "weight=100"
+```
+
+To activate the Blue/Green switch, we now only need to update the api:
+
+```bash
+# Switch the api from Blue to Green upstream, v1 -> v2
+$ curl -X PATCH http://kong:8001/apis/address-service \
+    --data "upstream_url=http://address.v2.service/address"
+```
+
+Incoming requests with host header set to `address.mydomain.com` will now be
+proxied by Kong to the new targets; 1/2 of the requests will go to
+`http://192.168.34.17:80/address` (`weight=100`), and the other 1/2 will go to
+`http://192.168.34.18:80/address` (`weight=100`).
+
+As always, the changes through the Kong management api are dynamic and will be
+effectuated immediately. No reload or restart is required, and no in progress
+requests will be dropped.
+
+[Back to Table of Contents](#table-of-contents)
+
+### **Canary releases**
+
+Using the ring-balancer, the target weights can be adjusted granularly and
+hence a smooth, controlled release can be done.
```

Comment:
"Target weights can be adjusted granularly, allowing for a smooth, controlled release"